kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java

```diff
@@ -122,18 +121,25 @@ private static void outputIndexInfo(CompilationDescription indexInfo) throws IOE
       if (outputFile.endsWith(IndexInfoUtils.KZIP_FILE_EXT)) {
         IndexInfoUtils.writeKzipToFile(indexInfo, outputFile);
       } else {
-        IndexInfoUtils.writeKindexToFile(indexInfo, outputFile);
+        System.err.printf("Unsupported output file: %s%n", outputFile);
+        System.exit(2);
       }
       return;
     }
 
     String outputDir = readEnvironmentVariable("KYTHE_OUTPUT_DIRECTORY");
-    if (Strings.isNullOrEmpty(System.getenv("KYTHE_INDEX_PACK"))) {
-      writeIndexInfoToFile(outputDir, indexInfo);
-      return;
-    }
-
-    new Archive(outputDir).writeDescription(indexInfo);
+    // Just rely on the underlying compilation unit's signature to get the filename, if we're not
+    // writing to a single kzip file.
+    String name =
+        indexInfo
+            .getCompilationUnit()
+            .getVName()
+            .getSignature()
+            .trim()
+            .replaceAll("^/+|/+$", "")
```

Comment:
Do these replaces create a risk of name collision?

Reply:
In theory maybe?  In practice, this is what we already used before.  So, might be ok.

Reply:
sgtm